### PR TITLE
Add definitions for `definitions.reference`

### PR DESCRIPTION
**Pull request by sdruskat**
**Related issues**

Refs: #206

Includes: #256, #261

**Describe the changes made in this pull request**

- Add definitions for `definitions.reference`
- Make some smaller fixes in schema

**Instructions to review the pull request**

- Look at `schema-guide.md`, again
- Check changes in `schema.json`

### CODE REVIEW

**Comment by jspaaks** [comment]
Took the liberty to merge 1.2.0 and 206-definitions-person into this for easier diff

File: schema-guide.md [hunk 1]
```diff
@@ -685,21 +948,21 @@ An entity can represent different types of entities, such as a team, an institut
         authors:
           - name: "The Research Software Project team"
         conference:
-          - name: "RC21 - Research Conference 2021"
+          name: "RC21 - Research Conference 2021"
```
Comment (jspaaks) [inline schema-guide.md:951]: good catch

File: schema-guide.md [hunk 2]
```diff
@@ -1776,1070 +2231,1287 @@ An entity can represent different types of entities, such as a team, an institut
...
     references:
-      - isbn: ...
+      - isbn: "9781603095075"
         type: generic
     ```
 
 ### `definitions.reference.issn`
 
-- **type**: `...`
+- **type**: `string` with pattern [`^\d{4}-\d{3}[\dxX]$`](https://regex101.com/library/jqobq9)
```
Comment (jspaaks) [inline schema-guide.md:2753]: didnt know this was possible, nice

File: schema-guide.md [hunk 3]
```diff
@@ -1776,1070 +2231,1287 @@ An entity can represent different types of entities, such as a team, an institut
...
-- **type**: `...`
+- **type**: Nonempty `string`
 - **required**: `false`
-- **description**: ...
+- **description**: The medium of the work.
 - **usage**:<br><br>
     ```yaml
     preferred-citation:
-      medium: ...
+      medium: "datasette"
```
Comment (jspaaks) [inline schema-guide.md:3010]: bit confusing maybe, perhaps we could use something like `Betamax` for humoristic purposes?

File: schema-guide.md [hunk 4]
```diff
@@ -1776,1070 +2231,1287 @@ An entity can represent different types of entities, such as a team, an institut
...
         type: generic
     ```
 
 ### `definitions.reference.notes`
 
-- **type**: `...`
+- **type**: Nonempty `string`
 - **required**: `false`
-- **description**: ...
+- **description**: Notes pertaining to the work.
```
Comment (jspaaks) [inline schema-guide.md:3057]: Should we mention that notes are not supposed to be picked up by downstream tooling (if that's even true)?

**Review by jspaaks — APPROVED** [review]
Nice work, I left some minor comments for your consideration

File: schema-guide.md (hunk 2 above)
Comment (sdruskat) [reply schema-guide.md:2753]: Innit! 👍

File: schema-guide.md (hunk 3 above)
Comment (sdruskat) [reply schema-guide.md:3010]: I see (also: includes a typo which makes `dataset` a substring, not good).
Using "5¼-inch floppy disk" now :).

File: schema-guide.md (hunk 4 above)
Comment (sdruskat) [reply schema-guide.md:3057]: Good idea.

Comment (sdruskat) [reply schema-guide.md:3057]: Good point! I don't think it's true for all tooling (reference managers may want to pick this up!), but some. Added https://github.com/citation-file-format/citation-file-format/pull/262/files#diff-28af06810cb7c393bb9ba0787f608e4f97918cc6b304541ba867916f243b4a38R3070-R3071. What do you think?

**Comment by sdruskat** [comment]
Thanks for the review @jspaaks! Can you quickly do a re-review of only https://github.com/citation-file-format/citation-file-format/pull/262#pullrequestreview-719396475 and https://github.com/citation-file-format/citation-file-format/pull/262#pullrequestreview-719397677? Many thanks!

**Review by jspaaks — APPROVED** [review]
floppy disk is ok, notes is not enough of a big deal to spend time on right now

**Comment by sdruskat** [comment]
Thanks for the review. Merging despite one failing check (failing to retrieve an action from the GH API, probably some downtime).

**Pull request merged**